llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -301,6 +301,27 @@ struct RISCVCallReturnHandler : public RISCVIncomingValueHandler {
 RISCVCallLowering::RISCVCallLowering(const RISCVTargetLowering &TLI)
     : CallLowering(&TLI) {}
 
+/// Return true if scalable vector with ScalarTy is legal for lowering.
+static bool isLegalElementTypeForRVV(Type *EltTy,
```

Comment:
Why do you need to filter the IR like this? I would hope anything unhandled would naturally fail and hit the fallback

Reply:
The other types aren't listed as legal in SelectionDAG so getRegisterTypeForCallingConv() returns the scalar type or something and I think bad things happened from there. SelectionDAG doesn't fail gracefully either.